### ISSUE TITLE
Installing ansible-lint 6.0 forces install of ansible-core 2.12.0

### ISSUE THREAD

**Issue by MallocArray** [ansible-community/ansible-lint#2014]
##### Summary
In previous versions, if I had Ansible 2.10.17 installed on the container, such as quay.io/ansible/ansible-runner:stable-2.10-devel I could then install ansible-lint using a pip3 requirements file and when it was installing 5.4.0, the process would leave the version of Ansible as it was

Since the release of 6.0.0, if I'm in the same base container and attempt to do a pip3 install ansible-lint, it automatically attempts to install ansible-core 2.12.3.  When done interactively the install fails with a warning about needing to uninstall ansible-base-2.10.  When done through Ansible Builder's Dockerfile, it completes successfully and installs ansible 2.12.3 which is not what I want at this time.

Unless ansible-lint is only compatible with ansible 2.12.x, it should not force the installation of 2.12

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
bash-4.4# ansible --version
ansible 2.10.17.post0
  config file = None
  configured module search path = ['/home/runner/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.8.12 (default, Sep 21 2021, 00:10:52) [GCC 8.5.0 20210514 (Red Hat 8.5.0-3)]


```

- ansible installation method: included in container image quay.io/ansible/ansible-runner:stable-2.10-devel
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

latest container image quay.io/ansible/ansible-runner:stable-2.10-devel

##### STEPS TO REPRODUCE

```console (paste below)
bash-4.4# ansible --version
ansible 2.10.17.post0
  config file = None
  configured module search path = ['/home/runner/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.8.12 (default, Sep 21 2021, 00:10:52) [GCC 8.5.0 20210514 (Red Hat 8.5.0-3)]
bash-4.4# pip3 install ansible-lint
Collecting ansible-lint
  Downloading ansible_lint-6.0.0-py3-none-any.whl (136 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 136.8/136.8 KB 4.5 MB/s eta 0:00:00
Collecting rich>=9.5.1
  Downloading rich-12.0.0-py3-none-any.whl (224 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 224.0/224.0 KB 39.0 MB/s eta 0:00:00
Collecting ansible-compat>=2.0.0
  Downloading ansible_compat-2.0.0-py3-none-any.whl (16 kB)
Collecting enrich>=1.2.6
  Downloading enrich-1.2.7-py3-none-any.whl (8.7 kB)
Collecting ansible-core>=2.12.0
  Downloading ansible-core-2.12.3.tar.gz (7.8 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.8/7.8 MB 96.8 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Collecting ruamel.yaml<0.18,>=0.15.34
  Downloading ruamel.yaml-0.17.21-py3-none-any.whl (109 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 109.5/109.5 KB 26.6 MB/s eta 0:00:00
Collecting wcmatch>=7.0
  Downloading wcmatch-8.3-py3-none-any.whl (42 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.9/42.9 KB 12.0 MB/s eta 0:00:00
Collecting yamllint>=1.25.0
  Downloading yamllint-1.26.3.tar.gz (126 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 126.7/126.7 KB 31.7 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Requirement already satisfied: pyyaml in /usr/lib64/python3.8/site-packages (from ansible-lint) (5.4.1)
Requirement already satisfied: packaging in /usr/local/lib/python3.8/site-packages (from ansible-lint) (21.3)
Collecting subprocess-tee>=0.3.5
  Downloading subprocess_tee-0.3.5-py3-none-any.whl (8.0 kB)
Requirement already satisfied: jinja2 in /usr/lib/python3.8/site-packages (from ansible-core>=2.12.0->ansible-lint) (2.10.3)
Requirement already satisfied: cryptography in /usr/lib64/python3.8/site-packages (from ansible-core>=2.12.0->ansible-lint) (2.8)
Collecting resolvelib<0.6.0,>=0.5.3
  Downloading resolvelib-0.5.4-py2.py3-none-any.whl (12 kB)
Collecting pygments<3.0.0,>=2.6.0
  Downloading Pygments-2.11.2-py3-none-any.whl (1.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.1/1.1 MB 57.1 MB/s eta 0:00:00
Collecting commonmark<0.10.0,>=0.9.0
  Downloading commonmark-0.9.1-py2.py3-none-any.whl (51 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 51.1/51.1 KB 14.9 MB/s eta 0:00:00
Collecting ruamel.yaml.clib>=0.2.6
  Downloading ruamel.yaml.clib-0.2.6-cp38-cp38-manylinux1_x86_64.whl (570 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 570.4/570.4 KB 75.0 MB/s eta 0:00:00
Collecting bracex>=2.1.1
  Downloading bracex-2.2.1-py3-none-any.whl (12 kB)
Collecting pathspec>=0.5.3
  Downloading pathspec-0.9.0-py2.py3-none-any.whl (31 kB)
Requirement already satisfied: setuptools in /usr/lib/python3.8/site-packages (from yamllint>=1.25.0->ansible-lint) (41.6.0)
Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /usr/local/lib/python3.8/site-packages (from packaging->ansible-lint) (3.0.7)
Requirement already satisfied: six>=1.4.1 in /usr/lib/python3.8/site-packages (from cryptography->ansible-core>=2.12.0->ansible-lint) (1.12.0)
Requirement already satisfied: cffi!=1.11.3,>=1.8 in /usr/lib64/python3.8/site-packages (from cryptography->ansible-core>=2.12.0->ansible-lint) (1.13.2)
Requirement already satisfied: MarkupSafe>=0.23 in /usr/lib64/python3.8/site-packages (from jinja2->ansible-core>=2.12.0->ansible-lint) (1.1.1)
Requirement already satisfied: pycparser in /usr/lib/python3.8/site-packages (from cffi!=1.11.3,>=1.8->cryptography->ansible-core>=2.12.0->ansible-lint) (2.19)
Using legacy 'setup.py install' for ansible-core, since package 'wheel' is not installed.
Using legacy 'setup.py install' for yamllint, since package 'wheel' is not installed.
Installing collected packages: resolvelib, commonmark, subprocess-tee, ruamel.yaml.clib, pygments, pathspec, bracex, yamllint, wcmatch, ruamel.yaml, rich, ansible-compat, enrich, ansible-core, ansible-lint
  Running setup.py install for yamllint ... done
  Running setup.py install for ansible-core ... error
  error: subprocess-exited-with-error
  
  × Running setup.py install for ansible-core did not run successfully.
  │ exit code: 1
  ╰─> [28 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-937dfud5/ansible-core_06bf18175f3a4ca6bc74c84e01529f7e/setup.py", line 120, in <module>
          _validate_install_ansible_core()
        File "/tmp/pip-install-937dfud5/ansible-core_06bf18175f3a4ca6bc74c84e01529f7e/setup.py", line 91, in _validate_install_ansible_core
          raise RuntimeError(
      RuntimeError:
      
          ****************************************************************************
      
          Cannot install ansible-core with a pre-existing ansible-base==2.10.17.post0
          installation.
      
          Installing ansible-core with ansible-2.9 or older, or ansible-base-2.10
          currently installed with pip is known to cause problems. Please uninstall
          ansible-base and install the new version:
      
              pip uninstall ansible-base
              pip install ansible-core
      
          If you want to skip the conflict checks and manually resolve any issues
          afterwards, set the ANSIBLE_SKIP_CONFLICT_CHECK environment variable:
      
              ANSIBLE_SKIP_CONFLICT_CHECK=1 pip install ansible-core
      
          ****************************************************************************
      
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> ansible-core

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```

##### Desired Behavior

Able to install ansible-lint 6.0.0 without upgrading ansible 2.10.17

##### Actual Behavior

Please give some details of what is actually happening.
Include a [minimum complete verifiable example] with:

after using Ansible builder to create an EE based on 2.10 with ansible-lint in the requirements.txt file, the resulting image has upgraded ansible

```paste below
ansible [core 2.12.3]
  config file = None
  configured module search path = ['/home/runner/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/site-packages/ansible
  ansible collection location = /home/runner/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.8.12 (default, Sep 21 2021, 00:10:52) [GCC 8.5.0 20210514 (Red Hat 8.5.0-3)]
  jinja version = 2.10.3
  libyaml = True

```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by cognifloyd** [comment]
Yes, ansible-lint 6.0 requires ansible-core 2.12+
https://github.com/ansible-community/ansible-lint/blob/f12a861c79f2714eba123391e54842a0197bfee5/setup.cfg#L67

ansible-lint 5.4 didn't declare its dependency on ansible-core directly, as it offered two extras to select ansible vs ansible-core. But, it really isn't an optional runtime dependency, so we simplified in 6.0. Also, increasing the version facilitated removing a lot of legacy compatibility code.

TL;DR if you need Ansible < 2.12, you need to use ansible-lint < 6.0.

**Comment by MallocArray** [comment]
@cognifloyd I didn't find the wording in the Release to be clear about that
`Remove support for ansible 2.9`
That sounds like only 2.9 compatibility was removed and does not seem to indicate that 2.10 and 2.11 are out as well.

I guess it falls under
`Made ansible-core a direct dependency`
but doesn't list which version minimum or that it is incompatible with less than 2.12
